### PR TITLE
#13221 [Robosaur] Split files command

### PR DESCRIPTION
**Pull request by daniel-jeremia**
## Issue

Robosaur needs support for splitting a single large csv file into multiple projects

## Changes
* Added a new handler called `splitDocumentHandler`
* Added a new schema in Config called `splitDocumentConfig`
* Added support for splitting a single large document into multiple folders in the specified result folder

### CODE REVIEW

**Review by jonathancesario — APPROVED** [review]
LGTM

**Comment by jonathancesario** [comment]
@darwin-gautalius could you please review this one since you will be more familiar with the code

**Review by darwin-gautalius — APPROVED** [review]

File: sample/split-document/config.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "datasaur": {
+    "host": "https://app.datasaur.ai",
+    "clientId": "<CLIENT_ID>",
+    "clientSecret": "<CLIENT_SECRET>"
+  },
+  "projectState": {
+    "source": "local",
+    "path": "sample/apply-tags/state/projectState.json"
+  },
```
Comment (darwin-gautalius) [inline sample/split-document/config.json:10]: oh btw, is this necessary?

File: src/config/schema/split-document-schema-validator.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  allErrors: true,
+});
+
+const SplitDocumentSchema: JSONSchemaType<SplitDocumentConfig> = {
+  type: 'object',
+  properties: {
+    path: { type: 'string' },
+    header: { type: 'boolean' },
+    linesPerFile: { type: 'number' },
+    filesPerFolder: { type: 'number' },
```
Comment (darwin-gautalius) [inline src/config/schema/split-document-schema-validator.ts:14]: should we validate the configuration here?
```suggestion
    linesPerFile: { type: 'number', minimum: 1 },
    filesPerFolder: { type: 'number', minimum: 1 },
```

File: sample/split-document/config.json (hunk 1 above)
Comment (jonathancesario) [reply sample/split-document/config.json:10]: ooh iya ga deng kan ga ada request API ya :thinking:

**Pull request merged**